### PR TITLE
consul-improvements - ability to reuse gossip-key

### PR DESCRIPTION
**Pull request by jharshman**
* Reuse gossip key if already present as a k8s secret

Useful when sidecars are using the gossip key to speak with the consul cluster.

@lachie83

### CODE REVIEW

**Comment by jharshman** [comment]
marked WIP. fixing some issues

**Comment by jharshman** [comment]
Removed WIP label & pending approval

File: stable/consul/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: consul
 home: https://github.com/hashicorp/consul
-version: 1.3.5
+version: 1.3.6
```
Comment (mattfarina) [inline stable/consul/Chart.yaml]: This is an API change so the version should be `2.0.0`

Comment (jharshman) [reply stable/consul/Chart.yaml]: How is this an API change?

**Comment by jharshman** [comment]
@mattfarina looks like some coms trouble with the service.  Ran into a very similar issue and it turned out to be a bad node. 

log:
```
I0501 17:43:39.896] Waiting for consul-9141-1-consul-0.consul-9141-1-consul to come up
I0501 17:43:39.896] ping: bad address 'consul-9141-1-consul-0.consul-9141-1-consul.presubmit-5302-9141.svc'
```

coming from:
https://github.com/jharshman/charts/blob/95b3250ecd1514d4d55ba62feeb6d0fdb5ecc27f/stable/consul/templates/consul-statefulset.yaml#L130-L136

**Pull request merged**